test/e2e/tests/dependencies.go

```diff
@@ -0,0 +1,186 @@
+
+	t := time.Now().UTC().Truncate(time.Second)
+
+	appStartdate := date.Time{Time: t}
+	appEndDate := date.Time{Time: t.Add(48 * time.Hour)}
+	passID := uuid.NewV4()
+	pass := uuid.NewV4()
+
+	fmt.Printf("appPass: %s\n", pass.String())
+
+	appName := conf.AROResourceGroup + "-" + conf.Cluster + conf.Location
+
+	app, err := dcs.appClientSet.Create(ctx, mgmtgraphrbac.ApplicationCreateParameters{
+		DisplayName:             to.StringPtr(appName),
+		Homepage:                to.StringPtr("https://" + appName),
+		IdentifierUris:          &[]string{"https://" + appName},
+		AvailableToOtherTenants: to.BoolPtr(false),
+		PasswordCredentials: &[]mgmtgraphrbac.PasswordCredential{
+			{
+				StartDate: &appStartdate,
+				EndDate:   &appEndDate,
+				KeyID:     to.StringPtr(passID.String()),
+				Value:     to.StringPtr(pass.String()),
+			},
+		},
+		ReplyUrls: &[]string{},
+	})
+
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	sp, err := dcs.spClientSet.Create(ctx, mgmtgraphrbac.ServicePrincipalCreateParameters{
+		AppID:          app.AppID,
+		AccountEnabled: to.BoolPtr(true),
+	})
+
+	if err != nil {
+		return nil, "", nil, err
+	}
+
+	return &app, pass.String(), &sp, err
+}
+
+func deleteServicePrincipal(ctx context.Context, appID string, spID string, dcs *DepClientSet) error {
+
+	_, err := dcs.appClientSet.Delete(ctx, appID)
+	return err
+
+}
+
+func NewDependencyClientSet(conf infra.EnvConfig) (*DepClientSet, error) {
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err != nil {
+		return nil, err
+	}
+
+	config := auth.NewClientCredentialsConfig(conf.AzureClientID, conf.AzureClientSecret, conf.AzureTenantID)
+	config.Resource = azure.PublicCloud.GraphEndpoint
+
+	rbAuthorizer, err := config.Authorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	return &DepClientSet{
+		rgClientSet:  features.NewResourceGroupsClient(conf.AzureSubscriptionID, authorizer),
+		appClientSet: graphrbac.NewApplicationsClient(conf.AzureTenantID, rbAuthorizer),
+		spClientSet:  graphrbac.NewServicePrincipalClient(conf.AzureTenantID, rbAuthorizer),
+	}, nil
+}
+
+func main() {
+	rand.Seed(time.Now().UnixNano())
+
+	conf := infra.NewEnvConf()
+	clients, err := NewDependencyClientSet(conf)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+
+	ctx := context.Background()
+
+	err = createResourceGroup(ctx, clients, conf)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	app, appSecret, sp, err := createServicePrincipal(ctx, clients, conf)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+
+	fmt.Printf("sp: %s\n", *sp.AppDisplayName)
+	fmt.Printf("sp: %s\n", *sp.ObjectID)
+	fmt.Printf("appId: %s\n", *app.AppID)
+	fmt.Printf("appSecret: %s\n", appSecret)
+
+	defer func() {
+		err = deleteServicePrincipal(ctx, "4f7fe6fb-5e7b-4271-9dbe-e35a7faa1129", "", clients)
+		if err != nil {
+			fmt.Println(err.Error())
+			return
+		}
+	}()
+
+	infra, err := infra.NewInfrastructure()
```

Comment:
Please use same pattern as in our code base (either deploy pkg or this:
resources: https://github.com/Azure/ARO-RP/pull/972/commits/bbcc1a44aceac4ff6a8cf50da4a2abe3d5871787#diff-3b0df476d21fbd8a8fbcc6a29fb8f9394992b0b298c5384abc3a3053b7c4b98fR191
and as in the same PR:
```
t := &arm.Template{
		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
		ContentVersion: "1.0.0.0",
		Parameters: map[string]*arm.TemplateParameter{
			"sas": {
				Type: "object",
			},
		},
		Resources: []*arm.Resource{
			... resources functions ...
		},
	}
	return m.deployARMTemplate(ctx, resourceGroup, "resources", t, map[string]interface{}{
		"sas": map[string]interface{}{
			"value": map[string]interface{}{
				"signedStart":         m.doc.OpenShiftCluster.Properties.Install.Now.Format(time.RFC3339),
				"signedExpiry":        m.doc.OpenShiftCluster.Properties.Install.Now.Add(24 * time.Hour).Format(time.RFC3339),
				"signedPermission":    "rl",
				"signedResourceTypes": "o",
				"signedServices":      "b",
				"signedProtocol":      "https",
			},
		},
	})
```